src/Slices/CreateInstance/UI/CreateInstance.tsx

```diff
@@ -36,9 +36,15 @@ export const CreateInstance: React.FC<Props> = ({ serviceEntity }) => {
     service_entity: serviceEntity.name,
   });
 
-  const onSubmit = async (attributes: InstanceAttributeModel) => {
+  const onSubmit = async (
+    attributes: InstanceAttributeModel,
+    dirtyInputs: string[],
+    setIsDirty: (values: string[]) => void
+  ) => {
+    setIsDirty([]);
```

Comment:
what if the submit fails? You should clear the isDirty only if the submit is okay

Reply:
if it fails we put these values again,, changing it to just clearing would cause the same issues as in canceling, as setState is async so redirect would take place first and that would trigger prompt

Reply:
So placing it in the else section before the redirect with an await doesn't work?

Reply:
No as it doesn't return Promise, but change to the state doesn't happen immediate, or can happen later, which in that case happens and redirect trigger prompt

Reply:
Okay, can you add it as a comment in the code ;)